### PR TITLE
Introduce url at error message at page.goto

### LINKED ISSUE #358: Introduce url at error message at page.goto
https://github.com/kblok/puppeteer/pull/2/commits/52d3bc05dea670e087afced923e288e4633c09b2

### PR DESCRIPTION
**Pull request by kblok**
Closes #358

### CODE REVIEW

File: lib/PuppeteerSharp/Page.cs [hunk 1]
```diff
@@ -1740,7 +1740,7 @@ await Task.WhenAll(Frames.Select(frame => frame.EvaluateExpressionAsync(expressi
 
             if (response.errorText != null)
             {
-                throw new NavigationException(response.errorText.ToString());
+                throw new NavigationException($"{response.errorText.ToString()} at {url}");
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: We could add an `Url` property on the `NavigationException` class

Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: Would you leave the url on the `Message` as well?

Comment (Meir017) [reply lib/PuppeteerSharp/Page.cs]: I think so, similar to how the [ArgumentException](https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/ArgumentException.cs#L77) works

**Comment by kblok** [comment]
Done @Meir017

**Comment by Meir017** [comment]
@kblok conflicts 😢

**Comment by kblok** [comment]
Done @Meir017

**Pull request merged**